src/main/java/org/folio/dcb/domain/mapper/TransactionMapper.java

```diff
@@ -8,22 +8,25 @@
 public class TransactionMapper {
 
   public TransactionEntity mapToEntity(String transactionId, DcbTransaction dcbTransaction) {
-    if(dcbTransaction == null || dcbTransaction.getItem() == null || dcbTransaction.getPatron() == null) {
+    if(dcbTransaction == null || dcbTransaction.getItem() == null || dcbTransaction.getPatron() == null || dcbTransaction.getPickup() == null) {
```

Comment:
the null validations should be applied before the method invocation.
sth like if(invalidContent(dcbTransaction)){
return null;
or even better throw an Exception
}
mapToEntity()-method invocation